packages/toolkit/src/view/settings/integrations/AvailableIntegration.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from "react";
+import Image from "next/image";
+import { Integration, IntegrationMethod, Nullable } from "instill-sdk";
+
+import { Button, Dialog, TabMenu, useToast } from "@instill-ai/design-system";
+
+import { useAddIntegrationConnection, useIntegration } from "../../../lib";
+import { ConnectionForm } from "./ConnectionForm";
+
+const METHOD_TITLE: Record<IntegrationMethod, string> = {
+  METHOD_DICTIONARY: "Dictionary",
+  METHOD_OAUTH: "oAuth",
+};
+
+export type AvailableIntegrationProps = {
+  integration: Integration;
+  accessToken: Nullable<string>;
+  enableQuery: boolean;
```

Comment:
@orangecoloured To avoid props drilling we can use the useInstillStore to get these two value